draft-ietf-quic-manageability.md

```diff
@@ -587,6 +589,48 @@ Work is currently underway in the TLS working group to encrypt the SNI in TLS
 would make SNI-based application identification impossible through passive
 measurement.
 
+### Extracting Server Name Indication (SNI) Information
+
+If the SNI is not encrypted it can be derived form the QUIC Initial packet 
+by calculating the Initial Secret to decrypt the packet payload and parse the 
+QUIC CRYPTO Frame with the TLS ClientHello.
+
+As both the initial salt for the Initial Secret as well as CRYPTO frame itself
+are version-specific, the first step is always to parse the version number
+(second to sixth byte of the long header). Note that only long header packets
+carry the version number, so it is necessary to also check the if first bit of
+the QUIC packet is set to 1, indicating a long header. 
+
+Note, that proprietary QUIC versions, that have been deployed before standardization,
+might not set the first bit in a QUIC long header packets to 1. To parse these
+versions example code is provided in the appendix (see {{sec-google-version}}),
+however, it is expected that these versions will gradually disappear over time.
+
+When the version has been identified as QUIC version 1, the packet type needs to be
+verified as an Initial packet by checking that the third and fourth bit of the header
+are both set to 0. Then the Client Destination Connection ID
+needs to be extracted to calculate the Initial Secret together with
+the version specific initial salt, as described in {{QUIC-TLS}}. The length of the
+connection ID is indicated in the 6th byte of the header followed by the connection
+ID itself.
+
+To determine the end of the header and find the start of the payload further
+the packet number length, the source connection ID length, as well as the token
+length need to be extracted. The packet number length is defined by the seventh
+and eight bits of the header as described in section 17.2. of {{QUIC-TRANSPORT}}.
+The source connection ID length is specified in the byte after the destination
+connection ID. And the token length, which follows the source connection ID,
+is a variable length integer as specified in section 16 of {{QUIC-TRANSPORT}}.
+
+Finally after decryption, the Initial Client packet can be parsed to detect the
+CRYPTO frame that contains the TLS Client Hello, which then can be respectively
+parsed similar as for all other TLS connections. The Initial client packet may
+contain other frames, so the first byte of each frame need to be checked to identify
+the frame type and the skip over the frame. Note that the length of the frames is 
+depended on the frame type. Usually for QUIC version 1, the packet is expected 
+to only carry the CRYPTO frame and optionally padding frames. However, padding
+which is one byte of zeros, may occur after or also before the CRYPO frame.
```

Comment:
```suggestion
which is one byte of zeros, may also occur before or after the CRYPTO frame.
```

having trouble parsing "padding which is one byte of zeroes" -- is the only allowable padding here exactly one byte long? haven't dug into this in a while.